js/blocked_number_controller.js

```diff
@@ -0,0 +1,85 @@
+/* global , Whisper, storage, ConversationController */
+/* global textsecure: false */
+
+/* eslint-disable more/no-then */
+
+// eslint-disable-next-line func-names
+(function() {
+    'use strict';
+
+    window.Whisper = window.Whisper || {};
+
+    const blockedNumbers = new Whisper.BlockedNumberCollection();
+    window.getBlockedNumbers = () => blockedNumbers;
+
+    window.BlockedNumberController = {
+      getAll() {
+        try {
+          this.load();
+        } catch (e) {
+          console.warn(e);
+        }
+        return blockedNumbers;
+      },
+      reset() {
+        blockedNumbers.reset([]);
+      },
+      load() {
+        window.log.info('BlockedNumberController: starting initial fetch');
+
+        if (blockedNumbers.length) {
+          throw new Error('BlockedNumberController: Already loaded!');
+        }
+
+        if (!storage) {
+          throw new Error('BlockedNumberController: Could not load blocked numbers');
+        }
+
+        // Add the numbers to the collection
+        const numbers = storage.getBlockedNumbers();
+        blockedNumbers.add(
+          numbers.map(number => ({ number }))
```

Comment:
what are the `()` around `{ number }` for ?

Reply:
it's there so the compiler knows to return a dictionary instead of treating it as a function block.

Reply:
Right, makes sense! Thanks for the explanation.